libs/vkd3d/command.c

```diff
@@ -8605,19 +8605,16 @@ static VkResolveModeFlagBits vk_resolve_mode_from_d3d12(D3D12_RESOLVE_MODE mode)
 static const struct vkd3d_format *d3d12_command_list_get_resolve_format(struct d3d12_command_list *list,
         const struct d3d12_resource *dst_resource, const struct d3d12_resource *src_resource, DXGI_FORMAT format)
 {
+    if (!format)
+      return NULL;
+
     /* Depth-stencil formats require special care since the app will need to
      * pass in the stencil plane as subresource 1, but may also use a format
      * for which we only have one aspect flag set. Just ignore the explicit
      * format in that case so that we always know the full aspect mask. */
-    if (dst_resource->format->vk_aspect_mask & (VK_IMAGE_ASPECT_DEPTH_BIT | VK_IMAGE_ASPECT_STENCIL_BIT))
-        return dst_resource->format;
-
-    if (dst_resource->format->type != VKD3D_FORMAT_TYPE_TYPELESS)
+    if (dst_resource->format->vk_aspect_mask == (VK_IMAGE_ASPECT_DEPTH_BIT | VK_IMAGE_ASPECT_STENCIL_BIT))
```

Comment:
Why did bitwise-and change to equal here?

Reply:
Doesn't technically change anything, but the whole `TYPELESS` thing mentioned in the comment only ever matters for images that have both aspects set.

Reply:
ok.